File: src/main/java/net/minestom/server/resourcepack/ResourcePackStatus.java
package net.minestom.server.resourcepack;

import net.minestom.server.entity.Player;

/**
 * Represents the result of {@link Player#setResourcePack(ResourcePack)} in
 * {@link net.minestom.server.event.player.PlayerResourcePackStatusEvent}.
 */
public enum ResourcePackStatus {

    SUCCESSFULLY_LOADED,
    DECLINED,
    FAILED_DOWNLOAD,
    ACCEPTED,
    DOWNLOADED,
    INVALID_URL,
    FAILED_RELOAD,
    DISCARDED,
}

File: src/main/java/net/minestom/server/resourcepack/ResourcePackStatus.kt
package net.minestom.server.resourcepack

import net.minestom.server.resourcepack.ResourcePack

/**
 * Represents the result of [Player.setResourcePack] in
 * [net.minestom.server.event.player.PlayerResourcePackStatusEvent].
 */
enum class ResourcePackStatus {
    SUCCESS, DECLINED, FAILED_DOWNLOAD, ACCEPTED
}